scalyr_agent/config_main.py

```diff
@@ -1427,7 +1428,22 @@ def make_symlink(source, target):
             default=False,
             help="Starts the agent if the conditional restart file marker exists.",
         )
-
+        # Special flag which is used by the Windows installer. We use it to indicate to this binary
+        # to fix up permissions for agent.json file and agent.d/ directory. This file is also used
+        # to create initial config by the install which means we can't correctly set those
+        # permissions inside the .wxs wix spec file.
+        # Right now it can only be used with "--init-config" flag on Windows.
+        parser.add_option(
+            "",
+            "--fix-config-permissions",
+            dest="fix_config_permissions",
+            action="store_true",
+            default=False,
+            help=(
+                "Fix permissions for agent.json file and agent.d/ directory and make sure it's. "
```

Comment:
I would mention in the help string that this only applies for windows.

Reply:
Technically, we could only add this option is platform is windows, WDYT?

I think it should be fairly safe.

Reply:
Made that change in - a0ee41bd.

When we check for that option we have a platform starts with win guard / check so it should be safe.